### ISSUE TITLE
[1.0.0-alpha.1] Dropdown migration instructions missing

### ISSUE THREAD

**Issue by jacobq** [Dogfalo/materialize#5353]
The [Dropdown section of the v1.0.0 upgrade guide](https://github.com/Dogfalo/materialize/blob/v1-dev/v1-upgrade-guide.md#dropdown) does not mention that drop-down buttons in v1.0.0 [need to use the `data-target` attribute](http://next.materializecss.com/dropdown.html) instead of [`data-activates`](http://materializecss.com/dropdown.html). 
BTW, this may be causing / related to #5349.

## Expected Behavior
Migration guide should document this breaking change.

## Current Behavior
Drop-down functionality does not work, and instead an error can be seen in the developer console:
```
Uncaught TypeError: Cannot read property 'tabIndex' of null
```
This appears to be caused by `this.id = M.getIdFromTrigger(el);` (in `Dropdown` constructor) getting `null` since `getIdFromTrigger` returns the `data-target` attribute

## Possible Solution
* Update migration guide to bring this to developers' attention
* Add `null` check to prevent this from throwing an error
* Print warning in console (at least in dev/unminified version) when required attributes are not found

## Steps to Reproduce (for bugs)
Just try to use `data-activates` instead of `data-target` in the example used on the documentation site. Here is a JSBin [demonstrating the problem](http://output.jsbin.com/kagokel) (I just now noticed that you prefer CodePen, so I will try to remember to use that for future issue submissions here.)

## Context
I use drop-down buttons in my (Ember + Electron based) app. While attempting to migrate to v1.0.0-alpha.1 (following the guide) I encountered this problem. It did not occur in the version I was using previously (v.0.99.0).

**Comment by jacobq** [comment]
Also, is it intentional that nav-bar drop-down buttons still appear [to use `data-activates`](http://next.materializecss.com/navbar.html#dropdown1)?

**Comment by acburst** [comment]
Fixed in b17de319
